### PR TITLE
Parenthetical operations work and pass all tests

### PR DESCRIPTION
**Pull request by riseofthesaber**

### CODE REVIEW

File: src/lib/lang/evaluators/order1/parenExpression.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    //for the paren expression we just need to go through everything within the parentheses and evaluate it FIRST before evaluating other things
+export function evalParenExpression(src: string, node: TreeCursor): number | Sequence{
+
+    // result can either evaluate to a flat number or be a sequence depending on the contents within the parentheses
+    let results: number|Sequence;
+
+    // there is only one child within the parentheses
+        //reminder: node.firstChild SETS node to the first child and it RETURNS a bool 
+        //it does not return the first child like one might initially think
+    node.firstChild()
```
Comment (TNThomas) [inline src/lib/lang/evaluators/order1/parenExpression.ts:15]: If this is false, someone wrote a `()`. While not particularly useful to write, it's probably a behavior worth handling explicitly. Off the top of my head, I'm thinking () -> 0. That way, say 1d() -> [0] and {()} -> [0]

File: src/lib/lang/evaluators/order1/parenExpression.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import type { TreeCursor } from '@lezer/common'
+import { Sequence } from '../../dataStructures'
+import { evalOrderLast } from ".."
+import { ErrorNodeError } from '../errors'
+
+    //for the paren expression we just need to go through everything within the parentheses and evaluate it FIRST before evaluating other things
+export function evalParenExpression(src: string, node: TreeCursor): number | Sequence{
+
+    // result can either evaluate to a flat number or be a sequence depending on the contents within the parentheses
+    let results: number|Sequence;
```
Comment (TNThomas) [inline src/lib/lang/evaluators/order1/parenExpression.ts:10]: Nitpick: Can we move this down closer to where it's used? That way it spends a minimal amount of time taking up space in memory.

**Review by TNThomas — CHANGES_REQUESTED** [review]
Needs unit tests, but only minor notes on committed files. Keep up the good work!

**Comment by riseofthesaber** [comment]
this pull request is outdated, the changes requested were made and parenthetical

**Comment by riseofthesaber** [comment]
Whoops, accidentally cut myself off there, I think I actually should keep this open, it looks like it has been keeping up to date. Anyway, the requested changes were made and automated tests have been created. code is functional, legible, and up to date, it just needs to be pulled

File: src/lib/lang/evaluators/order1/parenExpression.ts (hunk 2 above)
Comment (riseofthesaber) [reply src/lib/lang/evaluators/order1/parenExpression.ts]: nitpick fixed

File: src/lib/lang/evaluators/order1/parenExpression.ts (hunk 1 above)
Comment (riseofthesaber) [reply src/lib/lang/evaluators/order1/parenExpression.ts]: Unfortunately, node.firstChild would return true even if there was no child present.

**Review by TNThomas — APPROVED** [review]

**Pull request merged**